06-building-a-program-start-to-finish/solving-the-problem/output/app/class/invertedIndexClass.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * Inverted Index class
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * class constructor
+   * @constructor
+  */
+  constructor() {
+    this.indices = {};
+    this.indexedFiles = {};
+  }
+
+  readFile(fileToRead) {
+    const fileReader =  new FileReader();
+    // readFiles[filename] = [];
+    fileReader.readAsText(fileToRead);
+    return new Promise((resolve, reject) => {
+      fileReader.onload = () => {
+        try {
+          return resolve(JSON.parse(fileReader.result));
+        } catch (e) {
+          return reject('improperly formatted JSON file');
+        }
+      };
+    });
+  }
+
+  /**
+   * genereate indexes
+   * @function
+   * @param {Array} bookArray
+   * @param {string} title
+   * @returns {Object} indexes
+  */
```

Comment:
I believe you JSDocs can be more descriptive of what the function does, its parameters and what the returned value is. (Please address all JSDocs on this file)

Your JSDocs says your parameters are "bookArray" and "title", I don't see them used anywhere in your function.

What are you returning here? An object or a boolean?